api/catalog.go

```diff
@@ -81,6 +81,27 @@ type CatalogDeregistration struct {
 	Namespace  string `json:",omitempty"`
 }
 
+type CompoundServiceName struct {
```

Comment:
Why not just `ServiceName`, that way we keep parity with the internal structure?

Reply:
I was hesitant to use it because didn't really seem descriptive enough for external consumers. We have a CompoundServiceName function to generate ServiceName structs internally, so there's some consistency there.

I don't feel too strongly about it though 🤔

Reply:
Hmm, that seems fine to me then. I don't have a strong opinion either. 👍